rtl/eclrtl/eclrtl.cpp

```diff
@@ -4213,6 +4213,192 @@ unsigned rtlDelayReturn(unsigned value, unsigned sleepTime)
     return value;
 }
 
+//---------------------------------------------------------------------------
+
+// strptime and related functions
+
+inline bool readValue(unsigned & value, size32_t & _offset, size32_t lenStr, const char * str, unsigned max, bool spaceIsZero = false)
+{
+    unsigned total = 0;
+    unsigned offset = _offset;
+    if (lenStr - offset < max)
+        max = lenStr - offset;
+    unsigned i=0;
+    for (; i < max; i++)
+    {
+        char next = str[offset+i];
+        if (next >= '0' && next <= '9')
+            total = total * 10 + (next - '0');
+        else if (next == ' ' && spaceIsZero)
+            total = total * 10;
+        else
+            break;
+    }
+    if (i == 0)
+        return false;
+    value = total;
+    _offset = offset+i;
+    return true;
+}
+
+const char * const monthNames[12] = { "January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December" };
+
+inline bool matchString(unsigned & value, size32_t & strOffset, size32_t lenStr, const byte * str, unsigned num, const char * const * strings, unsigned minMatch)
+{
+    unsigned startOffset = strOffset;
+    for (unsigned i =0; i < num; i++)
+    {
+        const char * cur = strings[i];
+        unsigned offset = startOffset;
+        while (offset < lenStr)
+        {
+            byte next = *cur++;
+            if (!next || toupper(next) != toupper(str[offset]))
+                break;
+            offset++;
+        }
+        if (offset - startOffset >= minMatch)
+        {
+            value = i;
+            strOffset = offset;
+            return true;
+        }
+    }
+    return false;
+}
+
+//This implements a subset of the specifiers allowed for strptime
+//Another difference is it works on a string with a separate length
+ECLRTL_API const char * rltSimpleStrPTime(size32_t lenStr, const char * str, const char * format, struct tm * tm)
```

Comment:
minor: rlt should be rtl (**r**un**t**ime**l**ibrary).  You could have kept the original name since this function cannot be called from generated code.